holopy/core/tests/test_io.py

```diff
@@ -230,6 +230,22 @@ def __init__(self, a, b, c=3):
         # only a & c appear because b is None, d is not in __init__ signature
         assert yaml.dump(instantiated) == '!S\na: 1\nc: 3\n'
 
+    @attr("fast")
+    def test_custom_treatment_of_ndarrays(self):
+        zero_length_array = np.array(3)
+        one_length_array = np.array([3])
+        longer_array = 3 * np.ones(10, dtype=int)
+        multi_D_array = 3 * np.ones((2, 2), dtype=int)
+        self.assertEqual(yaml.dump(zero_length_array), yaml.dump(3))
+        self.assertEqual(yaml.dump(one_length_array), yaml.dump([3]))
+        self.assertEqual(yaml.dump(longer_array), yaml.dump([3] * 10))
+        self.assertEqual(yaml.dump(multi_D_array), yaml.dump([[3, 3], [3, 3]]))
+
+    @attr("fast")
+    def test_custom_treatment_of_numpy_ufuncs(self):
```

Comment:
likewise, it might be worth splitting this up into a "test_load" and test_dump", or alternatively just testing the round trip.

These are just suggestions, I don't think it's vital.